googlemock/include/gmock/gmock-matchers.h

```diff
@@ -4764,7 +4771,14 @@ class ExceptionMatcherImpl {
       } else {
         return true;
       }
-    } catch (const std::exception& err) {
+    } catch (
+        typename std::conditional<
```

Comment:
Instead of defining this logic inline, could you define an internal template using declaration for this type? And then I would add comments to explain what it's doing.

Thanks!

Reply:
Done 👍